### PR TITLE
[LLDB] Fix crash when using tab completion on class variables

### LINKED ISSUE #81536: lldb crashes when using tab completion.
lldb version: trunk.

a.cpp
```
class B {
  int x;
};
class A {
  B t;
  int t2;
};

int main(int argc, char* argv[]) {
  A a;
  return 0;
}
```

command:
`clang++ -g a.cpp && lldb a.out -o "b a.cpp:11" -o "r"`

Enter `a.t.` and use tab completion causes lldb to crash:
```
(lldb) p a.t.terminate called after throwing an instance of 'std::out_of_range'
  what():  basic_string::substr: __pos (which is 4) > this->size() (which is 3)
LLDB diagnostics will be written to /tmp/diagnostics-eaf5b2
Please include the directory content when filing a bug report
[1]    2833408 IOT instruction  lldb a.out -o "b a.cpp:11" -o "r"
```

### PR DESCRIPTION
**Pull request by svs-quic**
We weren't checking to see if the partial_path was empty before adding completions and this led to crashes when the class object and a variable both start with  the same substring.

Fixes [#81536](https://github.com/llvm/llvm-project/issues/81536)

### CODE REVIEW

**Comment by Michael137** [comment]
Change itself is fine. But could you please add a test in `lldb/test/API/commands/expression/completion/TestExprCompletion.py` or `lldb/test/API/functionalities/completion/TestCompletion.py`?

**Review by Michael137 — APPROVED** [review]
lgtm thanks!

**Comment by svs-quic** [comment]
> lgtm thanks!

Thanks. I dont think I have commit access . Would you be able to merge this for me?

File: lldb/test/API/functionalities/completion/TestCompletion.py [hunk 1]
```diff
@@ -60,10 +60,12 @@ def test_dwim_print(self):
...
     def do_test_variable_completion(self, command):
         self.complete_from_to(f"{command} fo", f"{command} fooo")
-        self.complete_from_to(f"{command} fooo.", f"{command} fooo.")
+        self.complete_from_to(f"{command} fooo.", f"{command} fooo.t")
+        self.complete_from_to(f"{command} fooo.t.", f"{command} fooo.t.x")
         self.complete_from_to(f"{command} fooo.dd", f"{command} fooo.dd")
 
-        self.complete_from_to(f"{command} ptr_fooo->", f"{command} ptr_fooo->")
+        self.complete_from_to(f"{command} ptr_fooo->", f"{command} ptr_fooo->t")
+        self.complete_from_to(f"{command} ptr_fooo->t", f"{command} ptr_fooo->t.x")
```
Comment (svs-quic) [inline lldb/test/API/functionalities/completion/TestCompletion.py:68]: This should have been ptr_fooo->t. 

Will fix this in a follow up patch

**Pull request merged**